vaadin-context-menu-flow/src/main/resources/META-INF/resources/frontend/contextMenuConnector.js

```diff
@@ -1,108 +1,114 @@
+          }
+          target.dispatchEvent(new CustomEvent('vaadin-context-menu-before-open', {detail: detail}));
+        }),
+
+        updateOpenOn: tryCatchWrapper(function(eventType) {
+          this.removeListener();
+          this.openOnEventType = eventType;
+
+          customElements.whenDefined('vaadin-context-menu').then(() => {
+            if (Gestures.gestures[eventType]) {
             Gestures.addListener(target, eventType, this.openOnHandler);
           } else {
             target.addEventListener(eventType, this.openOnHandler);
           }
         });
-      },
-
-      removeListener: function() {
-        if (this.openOnEventType) {
-          if (Gestures.gestures[this.openOnEventType]) {
-            Gestures.removeListener(target, this.openOnEventType, this.openOnHandler);
-          } else {
-            target.removeEventListener(this.openOnEventType, this.openOnHandler);
+        }),
+
+        removeListener: tryCatchWrapper(function() {
+          if (this.openOnEventType) {
+            if (Gestures.gestures[this.openOnEventType]) {
+              Gestures.removeListener(target, this.openOnEventType, this.openOnHandler);
+            } else {
+              target.removeEventListener(this.openOnEventType, this.openOnHandler);
+            }
           }
-        }
-      },
+        }),
 
-      openMenu: function(contextMenu) {
-        contextMenu.open(this.openEvent);
-      },
+        openMenu: tryCatchWrapper(function(contextMenu) {
+          contextMenu.open(this.openEvent);
+        }),
 
-      removeConnector: function() {
-        this.removeListener();
-        target.$contextMenuConnector = undefined;
-      }
+        removeConnector: tryCatchWrapper(function() {
+          this.removeListener();
+          target.$contextMenuConnector = undefined;
+        })
 
-    };
-  },
+      };
+    }),
 
-  generateItems: function(menu, appId, nodeId) {
-    menu._containerNodeId = nodeId;
+    generateItems: tryCatchWrapper(function(menu, appId, nodeId) {
+      menu._containerNodeId = nodeId;
 
-    const getContainer = function(nodeId) {
-      try {
-        return window.Vaadin.Flow.clients[appId].getByNodeId(nodeId);
-      } catch (error) {
-        console.error("Could not get node %s from app %s", nodeId, appId);
-        console.error(error);
-      }
-    };
+      const getContainer = function(nodeId) {
+        try {
+          return window.Vaadin.Flow.clients[appId].getByNodeId(nodeId);
+        } catch (error) {
+          console.error("Could not get node %s from app %s", nodeId, appId);
+          console.error(error);
+        }
+      };
 
-    const getChildItems = function(parent) {
-      const container = getContainer(parent._containerNodeId);
-      const items = container && Array.from(container.children).map(child => {
-        const item = {component: child, checked: child._checked};
+      const getChildItems = tryCatchWrapper(function(parent) {
```

Comment:
I think it's safe to remove `tryCatchWrapper` from here since this method is only called from this same scope (in `generateItems`) which is already wrapped with `tryCatchWrapper`.

Reply:
Valid point. Removed.